Make all tests to run during maven build

oleksiimiroshnyk opened the pull request:
- Use junit "vintage engine" as we still have old junit 3/4 tests and in classpath there is junit 5
- Add logback configs for test so it will be found, seems log4j2 is not picked up/selected regardless it is in classpath
- 
Fixes: BroadleafCommerce/QA#5076

Commit:
BroadleafCommerce/QA#5076
- Use junit "vintage engine" as we still have old junit 3/4 tests and in classpath there is junit 5
- Add logback configs for test so it will be found, seems log4j2 is not picked up/selected regardless it is in classpath

riteshadhikari17 commented:
This is the error in jenkins
```
[ERROR] org.broadleafcommerce.test.common.rule.MvelHelperTest.testMvelMethodOverloadFailureCase -- Time elapsed: 0.208 s <<< ERROR!
java.io.IOException: 
Error: Could not find or load main class org.broadleafcommerce.test.common.rule.MvelOverloadFailureReproduction
Caused by: java.lang.ClassNotFoundException: org.broadleafcommerce.test.common.rule.MvelOverloadFailureReproduction
```

Commit:
BroadleafCommerce/QA#5076
- just test to see what's going on in jenkins

Commit:
BroadleafCommerce/QA#5076
- just test to see what's going on in jenkins

Commit:
BroadleafCommerce/QA#5076
- just test to see what's going on in jenkins

Commit:
BroadleafCommerce/QA#5076
- just test to see what's going on in jenkins

Commit:
BroadleafCommerce/QA#5076
- just test to see what's going on in jenkins

Commit:
BroadleafCommerce/QA#5076
- just test to see what's going on in jenkins

Commit:
BroadleafCommerce/QA#5076
- jenkins build/surefire forks test execution and when we start another java process classpath is relative and seems working directory for that process is not correctly set(should be inherited from the parent process, as we don't specify anything) and it fails with class not found errors. Fixed that by calculating and using absolute paths

riteshadhikari17 committed:
Update engine to latest version

riteshadhikari17 commented:
The lines I am talking about are coming from MvelTestUtils#line136. 

By the way I am checking rest of the tests and integration still has issues for 7.0
```
[INFO] --- maven-surefire-plugin:3.1.2:test (default-test) @ integration ---
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
```

This is 6.2
```
[INFO] --- maven-surefire-plugin:2.22.2:test (default-test) @ integration ---
[INFO] Tests run: 167, Failures: 0, Errors: 0, Skipped: 0
```

Commit:
BroadleafCommerce/QA#5076
- Fix that integration(in integration module) tests(which are testng) are started by surefire pluging
- Fix that in integration module spock & junit tests are started with testng test suite
- Get rid of junit4 dependency in integration module and update to junit5 as seems spock/gmaven compiles to junit5
- Fix some integration tests mainly because of ordering(even with depends on group/method still have order variations) and because of removal of deprecated classes
- Fix integration tests running as seems spring application context caching gives problems because of our some static variables usage

Commit:
Merge branch 'develop-7.0.x' into QA-5076-not-all-tests-are-running-during-maven-build

riteshadhikari17 commented on common/pom.xml:
```diff
@@ -401,6 +416,7 @@
         <dependency>
             <groupId>org.hsqldb</groupId>
             <artifactId>hsqldb</artifactId>
+            <classifier>debug</classifier>
```
Is this classifier necessary?

oleksiimiroshnyk replied:
Not really, it helped me to identify the issue with tests/that spring testng integration creates multiple contexts...
Will remove it

riteshadhikari17 commented on common/src/test/java/org/broadleafcommerce/test/common/rule/MvelHelperTest.java:
```diff
@@ -40,6 +40,7 @@
 public class MvelHelperTest extends TestCase {
 
     private static final Log LOG = LogFactory.getLog(MvelHelperTest.class);
+    public static boolean something = false;
```
Looks like added for debugging. If not, maybe a meaningful var name.

riteshadhikari17 commented on common/src/test/resources/logback-test.xml:
```diff
@@ -0,0 +1,44 @@
+    <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <Pattern>
+                %d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n
```
I have this pattern for all modules. lets stay consistent
`[%-5level] %d{HH:mm:ss.SSS} [%t] %logger{36} - %msg%n`

riteshadhikari17 commented on integration/pom.xml:
```diff
@@ -58,7 +71,9 @@
                             <goal>test</goal>
                         </goals>
                         <configuration>
+<!--                            <argLine>-Dmaven.surefire.debug -Xmx1024m ${surefire.argLine}</argLine>-->
```
Can we delete these commented lines?

oleksiimiroshnyk replied:
I would like to keep them, just in case the next time will need to debug tests that are running through surefire plugin it could save time to find those arguments

riteshadhikari17 commented on integration/src/test/resources/logback-test.xml:
```diff
@@ -0,0 +1,43 @@
+    <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <Pattern>
+                %d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n
```
Same as above. pattern `[%-5level] %d{HH:mm:ss.SSS} [%t] %logger{36} - %msg%n`

riteshadhikari17 commented on pom.xml:
```diff
@@ -1285,6 +1291,7 @@
                 <groupId>org.hsqldb</groupId>
                 <artifactId>hsqldb</artifactId>
                 <version>${hsqldb.version}</version>
+                <classifier>debug</classifier>
```
same as above.

riteshadhikari17 commented on integration/pom.xml:
```diff
@@ -216,11 +235,12 @@
-        </dependency>
+        <!-- not adding junit for now, as we have old junit4
+            and spock/gmaven will compile with junit5 support
+            so rely on junit(junit-platform) that comes fomr spring/groovy
```
Typo. change `fomr` to `from`

riteshadhikari17 reviewed (changes requested):
For me I see a failure in `CatalogTest` locally (both IDE and shell) but I don't see that being run on jenkins build. Could you check what's going on? Weird thing is there are only 161 tests in older builds so it is picking up new things.
```
[ERROR] Tests run: 161, Failures: 1, Errors: 0, Skipped: 10, Time elapsed: 12.25 s <<< FAILURE! -- in TestSuite
[ERROR] org.broadleafcommerce.core.catalog.service.CatalogTest.testCatalog -- Time elapsed: 0.010 s <<< FAILURE!
java.lang.AssertionError
	at org.broadleafcommerce.core.catalog.service.CatalogTest.testCatalog(CatalogTest.java:132)
```

riteshadhikari17 commented:
![image](https://github.com/BroadleafCommerce/BroadleafCommerce/assets/35038090/5297fd36-22eb-4e68-880e-2869c7353539)

riteshadhikari17 commented:
It passes if I run the test individually but fails with I run whole module. It looks like it's using previous state because there is 't-shirt' as first category. `OrderTest` fails even if this passes.
![image](https://github.com/BroadleafCommerce/BroadleafCommerce/assets/35038090/f38d7bb9-0271-4c43-ba1d-170bbd1d030e)

Commit:
BroadleafCommerce/QA#5076
- clean up categories before test

oleksiimiroshnyk commented:
Hm, seems order of execution is different for some reason... added logic to clean up categories. Hope this will not break something else

riteshadhikari17 commented:
Merging this for now. Will handle local issue separately.
